[BUG] Searxng parse response from json to Java object error

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

## Describe the bug
<!-- A clear and concise description of what the bug is. -->

When using `langchain4j-community-web-search-engine-searxng` to search web content, the following error occur: 

```
java.lang.RuntimeException: 
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.lang.String` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 20703] (through reference chain: dev.langchain4j.community.web.search.searxng.SearXNGResponse["answers"]->java.util.ArrayList[0])
	at dev.langchain4j.community.web.search.searxng.SearXNGClient.search(SearXNGClient.java:73)
	at dev.langchain4j.community.web.search.searxng.SearXNGWebSearchEngine.search(SearXNGWebSearchEngine.java:88)
	at dev.langchain4j.web.search.WebSearchEngineIT.should_search_with_max_results(WebSearchEngineIT.java:50)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.lang.String` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 20703] (through reference chain: dev.langchain4j.community.web.search.searxng.SearXNGResponse["answers"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1767)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1541)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1446)
	at com.fasterxml.jackson.databind.DeserializationContext.extractScalarFromObject(DeserializationContext.java:958)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseString(StdDeserializer.java:1441)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:218)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:184)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2125)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1533)
	at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:33)
	at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:23)
	at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)
	at dev.langchain4j.community.web.search.searxng.SearXNGClient.search(SearXNGClient.java:70)
	... 5 more
```

Looks like the `answers` field in the `SearXNGResponse` is wrong format.

## Log and Stack trace
<!-- Please provide a log and a stack trace (with exception), if applicable. -->

As above.

## Minimal Example to Reproduce
<!-- Please provide a relevant code snippets to reproduce this bug. -->

Just search using `langchain4j-community-web-search-engine-searxng`

## Expected behavior
<!-- A clear and concise description of what you expected to happen. -->

## Please complete the following information:
- LangChain4j Community version: 1.0.0-alpha1
- LangChain4j Community Component you used (e.g. `langchain4j-community-clickhouse`): langchain4j-community-web-search-engine-searxng
- Java version: 17
- Spring Boot version (if applicable): 

## Additional context
<!-- Add any other context about the problem here. -->

/cc @bnayfeh

--- a/web-search-engines/langchain4j-community-web-search-engine-searxng/pom.xml
+++ b/web-search-engines/langchain4j-community-web-search-engine-searxng/pom.xml
@@ -85,6 +85,18 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.tinylog</groupId>
+            <artifactId>tinylog-impl</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.tinylog</groupId>
+            <artifactId>slf4j-tinylog</artifactId>
+            <scope>test</scope>
+        </dependency>
+
     </dependencies>
 
 </project>

--- a/web-search-engines/langchain4j-community-web-search-engine-searxng/src/main/java/dev/langchain4j/community/web/search/searxng/SearXNGResponse.java
+++ b/web-search-engines/langchain4j-community-web-search-engine-searxng/src/main/java/dev/langchain4j/community/web/search/searxng/SearXNGResponse.java
@@ -3,7 +3,6 @@ package dev.langchain4j.community.web.search.searxng;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
-
 import java.util.List;
 
 @JsonNaming(SnakeCaseStrategy.class)
@@ -13,7 +12,7 @@ class SearXNGResponse {
     private String query;
     private long numberOfResults;
     private List<SearXNGResult> results;
-    private List<String> answers;
+    private List<Answer> answers;
     private List<String> corrections;
     private List<String> suggestions;
     private List<List<String>> unresponsiveEngines;
@@ -31,7 +30,7 @@ class SearXNGResponse {
         return results;
     }
 
-    public List<String> getAnswers() {
+    public List<Answer> getAnswers() {
         return answers;
     }
 
@@ -46,5 +45,35 @@ class SearXNGResponse {
     public List<List<String>> getUnresponsiveEngines() {
         return unresponsiveEngines;
     }
-}
 
+    @JsonNaming(SnakeCaseStrategy.class)
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    static class Answer {
+
+        private String url;
+        private String template;
+        private String engine;
+        private List<String> parsedUrl;
+        private String answer;
+
+        public String getUrl() {
+            return url;
+        }
+
+        public String getTemplate() {
+            return template;
+        }
+
+        public String getEngine() {
+            return engine;
+        }
+
+        public List<String> getParsedUrl() {
+            return parsedUrl;
+        }
+
+        public String getAnswer() {
+            return answer;
+        }
+    }
+}

--- a/web-search-engines/langchain4j-community-web-search-engine-searxng/src/test/java/dev/langchain4j/community/web/search/searxng/SearXNGWebSearchEngineIT.java
+++ b/web-search-engines/langchain4j-community-web-search-engine-searxng/src/test/java/dev/langchain4j/community/web/search/searxng/SearXNGWebSearchEngineIT.java
@@ -1,10 +1,14 @@
 package dev.langchain4j.community.web.search.searxng;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 import dev.langchain4j.web.search.WebSearchEngine;
 import dev.langchain4j.web.search.WebSearchEngineIT;
 import dev.langchain4j.web.search.WebSearchOrganicResult;
 import dev.langchain4j.web.search.WebSearchRequest;
 import dev.langchain4j.web.search.WebSearchResults;
+import java.util.HashMap;
+import java.util.Map;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
@@ -14,11 +18,6 @@ import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
 import org.testcontainers.utility.DockerImageName;
 import org.testcontainers.utility.MountableFile;
-
-import java.util.HashMap;
-import java.util.Map;
-
-import static org.assertj.core.api.Assertions.assertThat;
 
 @Testcontainers
 class SearXNGWebSearchEngineIT extends WebSearchEngineIT {
@@ -35,6 +34,8 @@ class SearXNGWebSearchEngineIT extends WebSearchEngineIT {
     protected WebSearchEngine searchEngine() {
         return SearXNGWebSearchEngine.builder()
                 .baseUrl("http://" + searxng.getHost() + ":" + searxng.getMappedPort(8080))
+                .logRequests(true)
+                .logResponses(true)
                 .build();
     }
 

